### PR TITLE
GH-1163: Update Password UI

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-1163

### CODE REVIEW

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      <button
+        class="btn btn-block btn-secondary"
+        type="button"
+        (click)="onCancel()"
+      >
+        Cancel
+      </button>
+    </div>
+    <div class="col-lg-6 col-md-12">
+      <button class="btn btn-block btn-primary" type="submit">Save</button>
```
Comment (znikola) [inline projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.html]: On a mobile view, the buttons don't have space between them:
Screen Shot 2019-04-07 at 18 01 51

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    this.cancelled.emit();
+  }
+
+  private matchPassword(ac: FormGroup) {
+    if (ac.get('newPassword').value !== ac.get('newPasswordConfirm').value) {
+      return { NotEqual: true };
+    }
+  }
+
+  ngOnDestroy() {}
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.ts]: Not used.  Removed.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      oldPassword: this.form.value.oldPassword,
+      newPassword: this.form.value.newPassword,
+    });
+  }
+
+  onCancel(): void {
+    this.cancelled.emit();
+  }
+
+  private matchPassword(ac: FormGroup) {
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.ts]: Fixed.  See the new code.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
+<form
+  (submit)="onSubmit()"
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.html]: Fixed. Thanks, I didn't know that.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { CustomFormValidators } from '../../../../ui/validators/custom-form-validators';
+import { FormUtils } from '../../../../utils/forms/form-utils';
+@Component({
+  selector: 'cx-update-password-form',
+  templateUrl: './update-password-form.component.html',
+  styleUrls: ['./update-password-form.component.css'],
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.ts]: You are right, good catch.
Fixed.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password/update-password.component.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  RoutingService,
+  UserService,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-update-password',
+  templateUrl: './update-password.component.html',
+  styleUrls: ['./update-password.component.css'],
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password/update-password.component.ts]: You are right, good catch.
Fixed.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+      fixture.detectChanges();
+      component.onSubmit();
+
+      expect(component.form.valid).toBeTruthy();
+      expect(component.submited.emit).toHaveBeenCalled();
+    });
+  });
+
+  describe('onCancel', () => {
+    it('should be called call when the Cancel button is clicked', () => {
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.spec.ts]: Fixed.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+      clickSubmit();
+      fixture.detectChanges();
+      fixture.whenStable().then(() => {
+        expect(isFormControlDisplayingError('oldPassword')).toBeTruthy();
+        expect(isFormControlDisplayingError('newPassword')).toBeTruthy();
+        expect(isFormControlDisplayingError('newPasswordConfirm')).toBeFalsy();
+      });
+    });
+
+    it('should NOT display whan all field have valid valiues', () => {
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.spec.ts]: Thanks.  Fixed.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password/update-password.component.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+})
+export class UpdatePasswordComponent implements OnInit, OnDestroy {
+  subscription = new Subscription();
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password/update-password.component.ts]: It's not used outside the component class.  It's now private.

File: projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.html (hunk 1 above)
Comment (plabadie) [reply projects/storefrontlib/src/lib/my-account/update-password/components/update-password-form/update-password-form.component.html]: fixed.

**Pull request merged**